Airport-Managment-System/Services/FlightService.cs

```
@@ -0,0 +1,207 @@
+using Airport_Management_System.Entities;
+using Airport_Management_System.Helper;
+namespace Airport_Management_System.Services
+{
+    public class FlightService
+    {
+        private readonly List<Flight?> flights;
+        private  string csvFilePath = @"../../../Data/Flight.csv";
+
+        public FlightService()
+        {
+            try
+            {
+                this.flights = CsvHelperService.ReadFromCsv<Flight>(csvFilePath) ?? new List<Flight>();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error reading CSV file: {ex.Message}");
+                this.flights = new List<Flight>(); 
+            }
```

since your source of truth is the csv file, I don't think you should swallow the exception and continue running the app in case of failure, the whole app should stop if there is something wrong in reading the csv file, so removing the try catch block makes more sense to me, what do u think?

---

yes it was for catch the error that i was face (i remove it)